62.CAD/common.hlsl

```diff
@@ -148,9 +131,65 @@ struct PSInput
     void setBezierP0(float2 p0) { data2.xy = p0; }
     void setBezierP1(float2 p1) { data2.zw = p1; }
     
-    // data3 (zw reserved for later)
+    // data3 xy
     float2 getBezierP2() { return data3.xy; }
     void setBezierP2(float2 p2) { data3.xy = p2; }
+
+    // Curves are split in the vertex shader based on their tmin and tmax
+    // Min curve is smaller in the minor coordinate (e.g. in the default of y top to bottom sweep,
+    // curveMin = smaller x / left, curveMax = bigger x / right)
+    // TODO: possible optimization: passing precomputed values for solving the quadratic equation instead
+
+    // data2, data3, data4
+    float2 getCurveMinA() { return data2.xy; }
+    float2 getCurveMinB() { return data2.zw; }
+    float2 getCurveMinC() { return data3.xy; }
+    float2 getCurveMaxA() { return data3.zw; }
+    float2 getCurveMaxB() { return data4.xy; }
+    float2 getCurveMaxC() { return data4.zw; }
+    
+    void setCurveMinA(float2 p) { data2.xy = p; }
+    void setCurveMinB(float2 p) { data2.zw = p; }
+    void setCurveMinC(float2 p) { data3.xy = p; }
+    void setCurveMaxA(float2 p) { data3.zw = p; }
+    void setCurveMaxB(float2 p) { data4.xy = p; }
+    void setCurveMaxC(float2 p) { data4.zw = p; }
+
+    // interp_data5, interp_data6    
+
+    // A, B, C quadratic coefficients from the min & max curves,
+    // swizzled to the major cordinate and with the major UV coordinate subtracted
+    // These can be used to solve the quadratic equation
+    //
+    // a, b, c = curveMin.a,b,c()[major] - uv[major]
+    // 
+    // SolveQuadratic:
+    // det = b*b-4.f*a*c;
+    // rcp = 0.5f/a;
+    // detSqrt = sqrt(det)*rcp;
+    // tmp = b*rcp;
+    // res = float2(-detSqrt,detSqrt)-tmp;
+    //
+    // Collapsed version:
+    // detrcp2 = det * rcp * rcp
+    // brcp = b * rcp
+    //
+    // (In fragment shader)
+    // detSqrt = sqrt(detrcp2)
+    // res = float2(-detSqrt,detSqrt)-bRcp;
+    float getMinCurveDetRcp2() { return interp_data5.x; }
+    float getMinCurveBrcp() { return interp_data5.y; }
```

Comment:
Similar to https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/52/files#r1332647748
This could be a struct named `PrecomputedCurveBoxValues` and you set/get it.
Also in it's constructor you could write the precomputation logic